### PR TITLE
Scrap bugs rarity

### PR DESCRIPTION
**Pull request by erwanlbp**
In response to #6

I did just the bugs for a start, if it's accepted I'll try the other ones.

To scrap the rarity we need to go on every bug page so it takes longer to scrap.

The PR for sure needs a bit of rework, it was to start the thing. So please comment :)

### CODE REVIEW

File: scraper/scraper.js [hunk 1]
```diff
@@ -112,8 +113,29 @@ async function loadBugs() {
...
   });
 
-  await browser.close();
   const arr = JSON.parse(result);
+
+  for (bug of arr) {
+    await page.goto(`${RARITY_BASE_URL}${encodeURIComponent(bug.name.replace(' ', '_'))}`);
+    bug.rarity = await page.evaluate(() => {
+      const rarity = document.querySelector("[data-source=rarity] div").textContent;
+      if (rarity.includes("Common")) {
```
Comment (SawyerHood) [inline scraper/scraper.js:141]: I'm fine with keeping these values as either the names they are now or a number value between 1 -> 5. When displaying these we are probably just going to show either the text or if we are fancy a few star icons. I think we should start with just the words

**Comment by SawyerHood** [comment]
Thanks for taking this on. We might want to make the scraping a little more efficient. This is going to make it hit 100s of pages (if we enable it for fish as well). If we do this we will probably want to run multiple pages at once (this work can be done in parallel). I think we should rework this to do that before I merge in. Let me know if you need any help!

**Review by SawyerHood — CHANGES_REQUESTED** [review]
See above ^ I forgot to request changes

**Comment by erwanlbp** [comment]
> Thanks for taking this on. We might want to make the scraping a little more efficient. This is going to make it hit 100s of pages (if we enable it for fish as well). If we do this we will probably want to run multiple pages at once (this work can be done in parallel). I think we should rework this to do that before I merge in. Let me know if you need any help!

I changed it to run in parallel, but loading 80+ tabs in parallel take some times :joy: 
So I changed it to run in parallel by chunks.
I timed :
- 45s for 4 parallel chunks
- more than 50s for 2 chunks
- around 52s for 3 chunks
- around 46s for the sequential

My machine is not a beast and I'm not equipped with fiber.

What do you think ?

File: scraper/scraper.js [hunk 2]
```diff
@@ -112,13 +113,49 @@ async function loadBugs() {
...
+  await browser.close();
   await loadImages(arr);
   fs.writeFileSync("bugs.json", JSON.stringify(arr));
   return arr;
 }
 
+async function loadBugsRarity(bugs, arrayStart, arrayEnd, browser) {
+  for (let i = arrayStart; i < arrayEnd; i++) {
+    const bug = bugs[i];
+    console.log(`[${arrayStart},${arrayEnd}] [${i}] Scrap bug rarity ${bug.name}`);
```
Comment (SawyerHood) [inline scraper/scraper.js:136]: This should be "scrape" not scrap

**Review by SawyerHood — APPROVED** [review]
Sorry I've been busy with work! This looks like a good starting point. In the future we might want to find a better data source for all of the data (I actually was able to rip all of the assets straight from the game on my computer) so we don't have to wait so long to scrape everything. This looks awesome!

**Pull request merged**